src/web/transactions.rs

```diff
@@ -3,120 +3,95 @@ use std::collections::BTreeSet;
+    UnlimitedMetadata,
+};
+use iroha_data_model::transaction::{
+    error::model::TransactionRejectionReason, model::TransactionPayload,
 };
+
+use core::num::{NonZeroU32, NonZeroU64};
 use serde::Serialize;
 
 #[derive(Serialize)]
-#[serde(tag = "t", content = "c")]
-pub enum TransactionDTO {
-    Committed(CommittedTransactionDTO),
-    Rejected(RejectedTransactionDTO),
+pub struct TransactionDTO {
+    hash: SerScaleHex<HashOf<VersionedSignedTransaction>>,
+    block_hash: SerScaleHex<HashOf<CommittedBlock>>,
+    payload: TransactionPayloadDTO,
+    signatures: BTreeSet<SignatureDTO>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    rejection_reason: Option<SerScaleHex<TransactionRejectionReason>>,
 }
 
-impl TryFrom<TransactionValue> for TransactionDTO {
+impl TryFrom<TransactionQueryResult> for TransactionDTO {
     type Error = color_eyre::Report;
 
-    fn try_from(tx: TransactionValue) -> Result<Self> {
-        match tx {
-            TransactionValue::Transaction(versioned) => {
-                let tx = versioned.into_v1();
-                let base = TransactionBase::new(tx.hash(), tx.payload, tx.signatures)
-                    .wrap_err("Failed to make TransactionBase")?;
-                Ok(Self::Committed(CommittedTransactionDTO { base }))
-            }
-            TransactionValue::RejectedTransaction(versioned) => {
-                let tx = versioned.into_v1();
-                let hash = tx.hash();
-                let RejectedTransaction {
-                    payload,
-                    signatures,
-                    rejection_reason,
-                } = tx;
-                let base = TransactionBase::new(hash, payload, signatures)
-                    .wrap_err("Failed to make TransactionBase")?;
-                Ok(Self::Rejected(RejectedTransactionDTO {
-                    base,
-                    rejection_reason: rejection_reason.into(),
-                }))
-            }
-        }
+    fn try_from(tx_result: TransactionQueryResult) -> Result<Self> {
+        let TransactionValue { tx, error } = tx_result.transaction();
+        let block_hash = tx_result.block_hash();
+
+        Self::new(
+            tx.hash(),
+            block_hash,
+            tx.payload().clone(),
+            tx.signatures().clone(),
+            error.clone(),
+        )
+        .wrap_err("Failed to make TransactionDTO")
     }
 }
 
-#[derive(Serialize)]
-struct TransactionBase {
-    hash: SerScaleHex<HashOf<SignedTransaction>>,
-    block_hash: SerScaleHex<Hash>,
-    payload: TransactionPayloadDTO,
-    signatures: BTreeSet<SignatureDTO>,
-}
-
-impl TransactionBase {
-    fn new<T, U>(hash: HashOf<SignedTransaction>, payload: Payload, signatures: T) -> Result<Self>
-    where
-        T: IntoIterator<Item = U>,
-        U: Into<Signature>,
-    {
+impl TransactionDTO {
+    fn new(
+        hash: HashOf<VersionedSignedTransaction>,
+        block_hash: &HashOf<CommittedBlock>,
```

Comment:
```suggestion
        block_hash: HashOf<CommittedBlock>,
```

I don't see a reason why other fields are consumed, but `block_hash` is borrowed. Suggest consuming it as well for uniformity.